### PR TITLE
Update CI workflow for C++ coverage generation and testing

### PR DESCRIPTION
**Pull request by nagajaideep**
#### Brief description of what is fixed or changed
- Modified the CI workflow (`ci.yml`) to include steps for generating and uploading code coverage reports for the C++ modules.  
- Integrated `gcov` with necessary flags for accurate coverage measurement.  
- Ensured compatibility with existing build and test processes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,7 +37,29 @@ jobs:
       - name: Build package
         run: |
           python scripts/build/install.py
-
+      
+      - name: Build C++ Code with Coverage
+        run: |
+          mkdir -p build
+          cd build
+          cmake -DCMAKE_CXX_FLAGS="--coverage" ..
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:45]: This is not the right way. Nice attempt though.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,7 +37,29 @@ jobs:
       - name: Build package
         run: |
           python scripts/build/install.py
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:39]: Here, just do, `CXXFLAGS=--coverage CFLAGS=--coverage python scripts/build/install.py`.

Comment (czgdp1807) [reply .github/workflows/ci.yml:39]: The above should work.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -37,7 +37,29 @@ jobs:
...
+  
+      - name: Run C++ Tests
+        run: |
+          cd build
+          ./test_binary  
+        # Capture and Generating Coverage Report with lcov
+      - name: Capture Coverage Data with lcov
+        run: |
+          cd build
+          lcov --capture --directory . --output-file coverage.info
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:56]: Add `lcov` in https://github.com/codezonediitj/pydatastructs/blob/main/environment.yml under `myst_nb`.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nagajaideep) [reply .github/workflows/ci.yml]: removed this part of the code for now sir ,will work on it after the current issue

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nagajaideep) [reply .github/workflows/ci.yml]: made the changes in ci.yml file

File: .github/workflows/ci.yml (hunk 3 above)
Comment (nagajaideep) [reply .github/workflows/ci.yml:49]: working on this

Comment (nagajaideep) [reply .github/workflows/ci.yml:49]: added lcov under myst_nb in enviromental.yml file

**Comment by nagajaideep** [comment]
image
how to tackel this issue sir?based upon my understanding it is not able create it or not able to edit it ?any steps to resolve this issue sir?

File: environment.yml [hunk 4]
```diff
@@ -11,3 +11,4 @@ dependencies:
     - sphinx==5.0
     - sphinx-readable-theme==1.3.0
     - myst_nb==0.17.2
+    - lcov
```
Comment (czgdp1807) [inline environment.yml:14]: This will not work. `lcov` from `pip` doesn't seem to work.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -36,12 +36,22 @@ jobs:
...
-
+          CXXFLAGS=--coverage CFLAGS=--coverage python scripts/build/install.py
+# coverage tests     
       - name: Run tests
         run: |
           python -m pytest --doctest-modules --cov=./ --cov-report=xml -s
 
+      - name: Capture Coverage Data with lcov
+        run: |
+          cd build
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:47]:
```suggestion
          cd build
          sudo apt-get update
          sudo apt-get install -y lcov
```

Try the above.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 5 above)
Comment (nagajaideep) [reply .github/workflows/ci.yml:52]: yes sir issues is resolved

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -33,15 +33,32 @@ jobs:
...
+          sudo apt-get update
+          sudo apt-get install -y lcov
+          lcov --capture --directory . --output-file coverage.info
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:55]:
```suggestion
          lcov --capture --directory . --output-file coverage.info --no-external
```

If you check the logs, then you will C++'s internal headers are also covered. We only want PyDataStructs C++ source files.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -33,15 +33,32 @@ jobs:
...
+# coverage tests     
       - name: Run tests
         run: |
           python -m pytest --doctest-modules --cov=./ --cov-report=xml -s
 
+      - name: Capture Coverage Data with lcov
+        run: |
+          cd build
+          sudo apt-get update
+          sudo apt-get install -y lcov
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:54]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -33,15 +33,32 @@ jobs:
...
+          lcov --capture --directory . --output-file coverage.info --no-external
+  
+      - name: Generate HTML Coverage Report with genhtml
+        run: |
+          cd build
```
Comment (czgdp1807) [inline .github/workflows/ci.yml:59]:
```suggestion
```

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**